src/main/java/com/faforever/client/chat/CountryFlagService.java

```diff
@@ -35,6 +39,24 @@
         .map(url -> new Image(url.toString(), true));
   }
 
+  @Cacheable(COUNTRY_NAMES)
+  public List<String> getCountries(String startsWith) {
+    String[] countries = Locale.getISOCountries();
+    if(startsWith == null) {
+      return Arrays.asList(countries);
+    }
+
+    startsWith = startsWith.toLowerCase();
+
+    ArrayList<String> result = new ArrayList(countries.length);
+    for(String country : countries)  {
+      if(country.toLowerCase().startsWith(startsWith))
+        result.add(country);
+    }
+
+    return result;
+  }
```

Comment:
A lot of code for a simple thing:
```suggestion
  return Arrays.stream(Locale.getISOCountries())
        .filter(country -> startsWith == null || country.startsWith(startsWith))
        .collect(Collectors.toList());
```

Reply:
Java stream api

Reply:
Didn't know about the stream API. Thanks really helpful! Ended out using a method since there was a bit much logic for a one liner.